code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -126,12 +126,74 @@
 		B.transform = M
 		B.pixel_x = rand(-12, 12)
 		B.pixel_y = rand(-12, 12)
-	if(prob(33))
-		new/obj/item/shard(drop_location())
-	playsound(src, "shatter", 70, 1)
+	if(isGlass)
+		playsound(src, "shatter", 70, 1)
+		if(prob(33))
+			new/obj/item/shard(drop_location())
+	else
+		B.force = 0
+		B.throwforce = 0
+		B.desc = "A carton with the bottom half burst open. Might give you a papercut."
 	transfer_fingerprints_to(B)
 	qdel(src)
 
+/obj/item/reagent_containers/food/drinks/MouseDrop(atom/over, atom/src_location, atom/over_location, src_control, over_control, params)
+	var/mob/user = usr
+	if (!istype(src_location))
+		return
+	if (!user || user.incapacitated())
+		return
+	// Attempted drink sliding
+	if (locate(/obj/structure/table) in src_location)
+		//Are we an expert slider?
+		var/datum/action/innate/A = get_action_of_type(user, /datum/action/innate/drink_fling)
+		if(A?.active)
```

Comment:
Could you use some early return logic for indentation, if possible?

Reply:
done as best I can without breaking up the proc